src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/OwnerNames.kt

```diff
@@ -0,0 +1,182 @@
+        return objectTracker.getObjectRepresentation(obj)
+    }
+    // if the current owner is `this` - no owner needed
+    if (obj === shadowStackFrame.instance) return null
+    // lookup for the object in local variables and use the local variable name if found
+    shadowStackFrame.findLocalVariableReferringTo(obj)?.let { access ->
+        return if (isThisName(access.variableName)) null else access.variableName
+    }
+    // lookup for a field name in the current stack frame `this`
+    shadowStackFrame.findCurrentReceiverFieldReferringTo(obj)?.let { access ->
+        return if (isThisName(access.fieldName)) null else access.fieldName
+    }
+    // lookup for the constant referencing the object
+    constants[obj]?.let { return it }
+    // otherwise return object's string representation
+    return objectTracker.getObjectRepresentation(obj)
+}
+
+/**
+ * Finds the owner name of a given atomic API method call.
+ * Handles the following APIs:
+ * - `Atomic[Reference|Integer|Long|Boolean]`;
+ * - `Atomic[Reference|Integer|Long]Array`;
+ * - `Atomic[Reference|Integer|Long]FieldUpdater`;
+ * - `VarHandle`;
+ * - `Unsafe`.
+ *
+ * @param atomic the atomic API method receiver for which the owner name is to be found.
+ * @param arguments an array of arguments provided for the atomic operation.
+ * @param atomicMethodDescriptor a descriptor providing metadata about the atomic method being analyzed.
+ * @param shadowStackFrame the current shadow stack frame, representing the program's stack state during execution.
+ * @param objectTracker the object tracker, used to provide representations of tracked objects.
+ * @param constants a mapping from objects to their associated constant names.
+ * @return a pair consisting of the determined owner name as a string,
+ *   and a list of parameters associated with the atomic operation.
+ */
+internal fun findAtomicOwnerName(
+    atomic: Any,
+    arguments: Array<Any?>,
+    atomicMethodDescriptor: AtomicMethodDescriptor,
+    shadowStackFrame: ShadowStackFrame,
+    objectTracker: ObjectTracker,
+    constants: Map<Any, String>,
+): Pair<String, List<Any?>> {
+    val apiKind = atomicMethodDescriptor.apiKind
+    var ownerName: String? = null
+    var params: List<Any?>? = null
+
+    var arrayAccess = ""
+    if (apiKind == AtomicApiKind.ATOMIC_ARRAY) {
+        val info = atomicMethodDescriptor.getAtomicArrayAccessInfo(atomic, arguments)
+        arrayAccess = "[${(info.location as ArrayElementByIndexAccessLocation).index}]"
+        params = info.arguments
+    } else if (apiKind == AtomicApiKind.ATOMIC_OBJECT) {
+        params = arguments.asList()
+    }
+
+    fun getOwnerName(owner: Any?, className: String?, location: AccessLocation): String {
+        val owner = findOwnerName(owner, className, shadowStackFrame, objectTracker, constants)
+        return when (location) {
+            is ArrayElementByIndexAccessLocation -> {
+                (owner ?: "this") + "[${location.index}]"
```

Comment:
for lines:
> 113: arrayAccess = "[${(info.location as ArrayElementByIndexAccessLocation).index}]"

and

> 123: (owner ?: "this") + "[${location.index}]"

Will the result contain double brackets in the representation, since you return from this method a concatination `ownerName + arrayAccess`? Why is `arrayAccess` needed in general?
Does not seem very obvious to me. Could you please check that there is no mistake and add a clarifying comment here